panic: value is null during converting files from Prometheus stack, no clue for further yaml debug

I try to convert files from well-known repo for [Prometheus stack](https://github.com/prometheus-operator/kube-prometheus)

It mostly works fine, however three files raise an error. It would be nice to know at least, which line or tag in YAML file causes t. I will convert these files manually, no big deal, but probably you would like to have a look:

[prometheus-roleBindingSpecificNamespaces.yaml](https://github.com/prometheus-operator/kube-prometheus/blob/main/manifests/prometheus-roleBindingSpecificNamespaces.yaml):
```
panic: value is null

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.LengthInt(0x0, 0x0, 0x0, 0x0, 0xc00014e801)
	/Users/k2m30/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1058 +0x28b
github.com/zclconf/go-cty/cty.Value.AsValueMap(0x0, 0x0, 0x0, 0x0, 0x13627e0)
	/Users/k2m30/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1315 +0x9d
main.yamlToHCL(0x1236b48, 0xc0001376a0, 0x11c5e20, 0xc00016a120, 0x0, 0x0, 0x0, 0x0, 0x1236b48, 0xc0001376a0, ...)
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:90 +0x151
main.YAMLToTerraformResources(0x12340d0, 0xc0000b0680, 0x0, 0x0, 0xc0000b0000, 0x0, 0x0, 0xc0000ae9ba, 0xc000099cd0)
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:162 +0x409
main.main()
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:204 +0x331
```


[prometheus-roleSpecificNamespaces.yaml](https://github.com/prometheus-operator/kube-prometheus/blob/main/manifests/prometheus-roleSpecificNamespaces.yaml):
```
panic: value is null

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.LengthInt(0x0, 0x0, 0x0, 0x0, 0xc000195901)
	/Users/k2m30/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1058 +0x28b
github.com/zclconf/go-cty/cty.Value.AsValueMap(0x0, 0x0, 0x0, 0x0, 0x13627e0)
	/Users/k2m30/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1315 +0x9d
main.yamlToHCL(0x1236b48, 0xc0001c2fd0, 0x11c5e20, 0xc0001e2d20, 0x0, 0x0, 0x0, 0x0, 0x1236b48, 0xc0001c2fd0, ...)
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:90 +0x151
main.YAMLToTerraformResources(0x12340d0, 0xc000010690, 0x0, 0x0, 0xc000010000, 0x0, 0x0, 0xc00001cb4a, 0xc00005fce0)
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:162 +0x409
main.main()
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:204 +0x331
```


[grafana-dashboardDefinitions.yaml](https://github.com/prometheus-operator/kube-prometheus/blob/main/manifests/grafana-dashboardDefinitions.yaml):
```
panic: value is null

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.LengthInt(0x0, 0x0, 0x0, 0x0, 0xc000128f01)
	/Users/k2m30/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1058 +0x28b
github.com/zclconf/go-cty/cty.Value.AsValueMap(0x0, 0x0, 0x0, 0x0, 0x13627e0)
	/Users/k2m30/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1315 +0x9d
main.yamlToHCL(0x1236b48, 0xc00027d0a0, 0x11c5e20, 0xc000ff0ff0, 0x0, 0x0, 0x0, 0x0, 0x1236b48, 0xc00027d0a0, ...)
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:90 +0x151
main.YAMLToTerraformResources(0x12340d0, 0xc00012a680, 0x0, 0x0, 0xc000120000, 0x0, 0x0, 0xc0001289ba, 0xc000111cd0)
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:162 +0x409
main.main()
	/Users/k2m30/go/pkg/mod/github.com/jrhouston/tfk8s@v0.1.5/tfk8s.go:204 +0x331
```

Thanks

--- a/tfk8s.go
+++ b/tfk8s.go
@@ -81,36 +81,56 @@ func stripServerSideFields(doc cty.Value) cty.Value {
 	return cty.ObjectVal(m)
 }
 
+// snakify converts "a-String LIKE this" to "a_string_like_this"
+func snakify(s string) string {
+	re := regexp.MustCompile(`\W`)
+	return strings.ToLower(re.ReplaceAllString(s, "_"))
+}
+
 // yamlToHCL converts a single YAML document Terraform HCL
 func yamlToHCL(doc cty.Value, providerAlias string, stripServerSide bool, mapOnly bool) (string, error) {
-	var name, resourceName string
 	m := doc.AsValueMap()
-	kind := m["kind"].AsString()
-	if kind != "List" {
-		metadata := m["metadata"].AsValueMap()
-		name = metadata["name"].AsString()
-		re := regexp.MustCompile(`\W`)
-		name = strings.ToLower(re.ReplaceAllString(name, "_"))
-		resourceName = strings.ToLower(kind) + "_" + name
-	} else if !mapOnly {
-		return "", fmt.Errorf("Converting v1.List to a full Terraform configuation is currently not supported")
+	docs := []cty.Value{doc}
+	if strings.HasSuffix(m["kind"].AsString(), "List") {
+		docs = m["items"].AsValueSlice()
 	}
 
-	if stripServerSide {
-		doc = stripServerSideFields(doc)
-	}
-	s := terraform.FormatValue(doc, 0)
-
-	var hcl string
-	if mapOnly {
-		hcl = fmt.Sprintf("%v\n", s)
-	} else {
-		hcl = fmt.Sprintf("resource %q %q {\n", resourceType, resourceName)
-		if providerAlias != "" {
-			hcl += fmt.Sprintf("  provider = %v\n\n", providerAlias)
+	hcl := ""
+	for i, doc := range docs {
+		mm := doc.AsValueMap()
+		kind := mm["kind"].AsString()
+		metadata := mm["metadata"].AsValueMap()
+		name := metadata["name"].AsString()
+		var namespace string
+		if v, ok := metadata["namespace"]; ok {
+			namespace = v.AsString()
 		}
-		hcl += fmt.Sprintf("  manifest = %v\n", strings.ReplaceAll(s, "\n", "\n  "))
-		hcl += fmt.Sprintf("}\n")
+
+		resourceName := kind
+		if namespace != "" && namespace != "default" {
+			resourceName = resourceName + "_" + namespace
+		}
+		resourceName = resourceName + "_" + name
+		resourceName = snakify(resourceName)
+
+		if stripServerSide {
+			doc = stripServerSideFields(doc)
+		}
+		s := terraform.FormatValue(doc, 0)
+
+		if mapOnly {
+			hcl += fmt.Sprintf("%v\n", s)
+		} else {
+			hcl += fmt.Sprintf("resource %q %q {\n", resourceType, resourceName)
+			if providerAlias != "" {
+				hcl += fmt.Sprintf("  provider = %v\n\n", providerAlias)
+			}
+			hcl += fmt.Sprintf("  manifest = %v\n", strings.ReplaceAll(s, "\n", "\n  "))
+			hcl += fmt.Sprintf("}\n")
+		}
+		if i != len(docs)-1 {
+			hcl += "\n"
+		}
 	}
 
 	return hcl, nil

--- a/tfk8s_test.go
+++ b/tfk8s_test.go
@@ -98,6 +98,83 @@ resource "kubernetes_manifest" "configmap_two" {
 	assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(output))
 }
 
+func TestYAMLToTerraformResourcesList(t *testing.T) {
+	yaml := `---
+apiVersion: v1
+kind: ConfigMapList
+items:
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: one
+  data:
+    TEST: one
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: two
+  data:
+    TEST: two
+- apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: two
+    namespace: othernamespace
+  data:
+    TEST: two
+`
+
+	r := strings.NewReader(yaml)
+	output, err := YAMLToTerraformResources(r, "", false, false)
+
+	if err != nil {
+		t.Fatal("Converting to HCL failed:", err)
+	}
+
+	expected := `
+resource "kubernetes_manifest" "configmap_one" {
+  manifest = {
+    "apiVersion" = "v1"
+    "data" = {
+      "TEST" = "one"
+    }
+    "kind" = "ConfigMap"
+    "metadata" = {
+      "name" = "one"
+    }
+  }
+}
+
+resource "kubernetes_manifest" "configmap_two" {
+  manifest = {
+    "apiVersion" = "v1"
+    "data" = {
+      "TEST" = "two"
+    }
+    "kind" = "ConfigMap"
+    "metadata" = {
+      "name" = "two"
+    }
+  }
+}
+
+resource "kubernetes_manifest" "configmap_othernamespace_two" {
+  manifest = {
+    "apiVersion" = "v1"
+    "data" = {
+      "TEST" = "two"
+    }
+    "kind" = "ConfigMap"
+    "metadata" = {
+      "name" = "two"
+      "namespace" = "othernamespace"
+    }
+  }
+}`
+
+	assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(output))
+}
+
 func TestYAMLToTerraformResourcesProviderAlias(t *testing.T) {
 	yaml := `---
 apiVersion: v1

